Source/WebCore/svg/SVGFEConvolveMatrixElement.cpp

```diff
@@ -216,54 +216,47 @@ void SVGFEConvolveMatrixElement::svgAttributeChanged(const QualifiedName& attrNa
 
 RefPtr<FilterEffect> SVGFEConvolveMatrixElement::createFilterEffect(const FilterEffectVector&, const GraphicsContext&) const
 {
-    int orderXValue = orderX();
-    int orderYValue = orderY();
-    if (!hasAttribute(SVGNames::orderAttr)) {
-        orderXValue = 3;
-        orderYValue = 3;
+    int orderXValue = 3;
+    int orderYValue = 3;
+    if (hasAttribute(SVGNames::orderAttr)) {
+        orderXValue = orderX();
+        orderYValue = orderY();
+        // Spec says order must be > 0. Bail if it is not.
+        if (orderXValue <= 0 || orderYValue <= 0)
+            return nullptr;
     }
-    // Spec says order must be > 0. Bail if it is not.
-    if (orderXValue < 1 || orderYValue < 1)
-        return nullptr;
```

Comment:
After reviewing this more carefully, this logic isn't equivalent either. We were early returning unconditionally before, now we're only early returning when the attribute is set.

Can you move out the early return of the `hasAttribute(SVGNames::orderAttr))` check?

The code should look like this:

```
     int orderXValue = 3;
     int orderYValue = 3;
     if (hasAttribute(SVGNames::orderAttr)) {
         orderXValue = orderX();
         orderYValue = orderY();
     }
     // Spec says order must be > 0. Bail if it is not.
     if (orderXValue <= 0 || orderYValue <= 0)
         return nullptr;
```

Reply:
Actually it is not, because it is impossible to return when the condition is not set.

3 <= 0 is always false.

Reply:
If the attribute is not set, we will NEVER return unless some outside force messes with the memory or CPU or something, which even in this case we don't want to do.

Reply:
Ah right, I missed that.

Reply:
Fwiw, in terms of performance, I don't think this really matters, given the compiler should be smart enough to optimize this.